### PR TITLE
Feature/eaa 472 build initial data access layer

### PR DESCRIPTION
**Pull request by Bartonkeys**

### CODE REVIEW

File: Beis.Ebss.CoreLayer/Beis.Ebss.CoreLayer.csproj [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
```
Comment (willappleby) [inline Beis.Ebss.CoreLayer/Beis.Ebss.CoreLayer.csproj:6]: If we're going to enable nullable types we should do this in all the projects for consistency - I think some of them have this disabled?

Comment (Bartonkeys) [reply Beis.Ebss.CoreLayer/Beis.Ebss.CoreLayer.csproj:6]: Do you want me to remove?

File: Beis.Ebss.ApplicationPortal.Web/Program.cs [hunk 2]
```diff
@@ -64,7 +70,11 @@
                   .SetSlidingExpiration(TimeSpan.FromMinutes(SessionTimeOutMinutes)));
 builder.Services.AddStackExchangeRedisCache(options => { options.Configuration = builder.Configuration["RedisPrimaryConnectionString"]; });
 
-//builder.Services.AddDbContext<HtgVendorSmeDbContext>(options => options.UseNpgsql(configuration["HelpToGrowDbConnectionString"]));
+builder.Services.AddDbContext<EbssContext>(options => options.UseNpgsql(builder.Configuration["EbssConnectionString"]));
```
[the remark this answers is not in the document]
Comment (Bartonkeys) [reply Beis.Ebss.ApplicationPortal.Web/Program.cs:73]: Format is:
`"EbssConnectionString": "Host=localHost:5432;Username=user;Password=pw;Database=ebss"`
I have passed onto Paul

File: Beis.Ebss.CoreLayer/Beis.Ebss.CoreLayer.csproj (hunk 1 above)
Comment (willappleby) [reply Beis.Ebss.CoreLayer/Beis.Ebss.CoreLayer.csproj:6]: I don't have a strong preference either way but think it's important to be consistent across the solution so either disable it here or enable in other projects.

File: Beis.Ebss.DataLayer/Entities/Address.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Beis.Ebss.DataLayer.Entities;
+
+public class Address: BaseEntity
+{
+    public string? UPRN { get; set; }
+    public string? AddressLine1 { get; set; }
+    public string? AddressLine2 { get; set; }
+    public string? Town { get; set; }
+    public string? County { get; set; }
+    public string? Postcode { get; set; }
```
Comment (spotla4) [inline Beis.Ebss.DataLayer/Entities/Address.cs:10]: We may want to have some of the address fields like Postcode not nullable types.

Comment (Bartonkeys) [reply Beis.Ebss.DataLayer/Entities/Address.cs:10]: Thanks Sisira, I will check this with Andy

**Review by spotla4 — APPROVED** [review]

File: Beis.Ebss.DataLayer/Entities/Address.cs (hunk 3 above)
Comment (Bartonkeys) [reply Beis.Ebss.DataLayer/Entities/Address.cs:10]: Ok to leave this as is. I made some chanages to bank details though to ensure account number and sort code are ints.

**Review by willappleby — APPROVED** [review]

**Pull request merged**